### PR TITLE
Test measures

### PR DESCRIPTION
**Pull request by LFISHER7**
We refactored `measures.py` in https://github.com/opensafely-core/interactive-templates/pull/87. This PR extends the existing tests to cover the rest of the module

### CODE REVIEW

**Review by bloodearnest — APPROVED** [review]

File: interactive_templates/templates/v2/tests/test_measures.py [hunk 1]
```diff
@@ -24,3 +27,129 @@ def test_redact_and_round_column(data_frame):
...
+    exp = pd.DataFrame(
+        {
+            "A": [1],
+            "B": ["a"],
+            "C": [10],
+        }
+    )
+
+    obs = measures.filter_data(filter_data_df, filters)
+    assert exp.equals(obs)
```
Comment (iaindillingham) [inline interactive_templates/templates/v2/tests/test_measures.py:59]: If the test fails, then you get a nicer message with `pandas.testing.assert_frame_equal`.

File: interactive_templates/templates/v2/tests/test_measures.py [hunk 2]
```diff
@@ -24,3 +27,129 @@ def test_redact_and_round_column(data_frame):
...
+
+
+@st.composite
+def input_df(draw):
+    nrows = draw(st.integers(min_value=20, max_value=20))
```
Comment (iaindillingham) [inline interactive_templates/templates/v2/tests/test_measures.py:64]: Isn't this always going to draw `20`?

File: interactive_templates/templates/v2/tests/test_measures.py [hunk 3]
```diff
@@ -24,3 +27,129 @@ def test_redact_and_round_column(data_frame):
...
+
+
+@given(df=input_df())
+def test_calculate_total_counts(df):
+    date = "2022-01-01"
+    df["date"] = date
+
+    obs = measures.calculate_total_counts(df, date, group="total", group_value="total")
+
+    # assert that both the group and group_value are "total" for every row
```
Comment (iaindillingham) [inline interactive_templates/templates/v2/tests/test_measures.py:103]: I don't think this comment is very useful; it says the same as the code, in English.

**Review by iaindillingham — APPROVED** [review]
Several comments. It's hard to review this thoroughly, as I'd need to unpick both the implementation and the test code. If I could make one suggestion, it would be to eschew testing the shape of a data frame in favour of, say, its values.

**Comment by LFISHER7** [comment]
Thanks @iaindillingham these are useful suggestions. I have a slight bias to checking shape having worked a lot with numpy, but can see it's a lot better to check more than this working with pandas.

**Pull request merged**